dashboard/lib/expired_deleted_account_pii_scrubber.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'stringio'
+require 'cdo/aws/metrics'
+require 'cdo/aws/s3'
+require 'cdo/chat_client'
+require 'cdo/honeybadger'
+
+# Queries for accounts soft-deleted at least 28 days ago and scrub them
+# of PII (personally identifiable information).
+#
+# This renders the accounts unrecoverable but retains as much useful non-PII
+# data as possible.
+#
+# Logs activity to the user-accounts and cron-daily Slack channels, as well as Cloudwatch.
+class ExpiredDeletedAccountPiiScrubber
+  class SafetyConstraintViolation < RuntimeError; end
+
+  attr_reader :dry_run, :scrub_accounts_deleted_since, :max_accounts_to_scrub, :num_accounts_scrubbed, :num_errors, :start_time
+  alias :dry_run? :dry_run
+
+  LOGGING_NAMESPACE = 'Platform/PiiScrubber'
+
+  def initialize(options = {})
+    @dry_run = options[:dry_run].nil? ? false : options[:dry_run]
+    raise ArgumentError.new('dry_run must be boolean') unless [true, false].include? @dry_run
+
+    # The amount of time after being soft-deleted that an account should be scrubbed of PII.
+    @scrub_accounts_deleted_since = options[:scrub_accounts_deleted_since] || 28.days.ago
+    raise ArgumentError.new('scrub_accounts_deleted_since must be Time') unless @scrub_accounts_deleted_since.is_a? Time
+
+    # Maximum number of accounts to scrub in a single run.
+    # This is a safety limit to prevent accidental deletion of too many accounts.
+    @max_accounts_to_scrub = options[:max_accounts_to_scrub] || 8000
+    raise ArgumentError.new('max_accounts_to_scrub must be Integer') unless @max_accounts_to_scrub.is_a? Integer
+
+    reset_metrics
+  end
+
+  def scrub_pii_from_expired_deleted_accounts!
+    reset_metrics
+
+    accounts_to_scrub.in_batches(of: 1000).each do |batch|
+      batch.each do |user|
+        scrub_user(user)
+        @num_accounts_scrubbed += 1
+      rescue StandardError => exception
+        @num_errors += 1
+        Honeybadger.notify(exception, context: {user_id: user.id})
+        log_message("Error scrubbing user_id #{user.id}: #{exception.message}")
+      end
+    end
+
+    if dry_run?
+      log_message("Dry run complete: would scrub #{@num_accounts_scrubbed} accounts. Encountered #{@num_errors} errors.")
+    else
+      log_message(format("Scrubbed #{@num_accounts_scrubbed} accounts in %.2f seconds. Encountered #{@num_errors} errors.", (Time.now - @start_time)))
+      upload_metrics
+    end
+
+    log_to_slack(summary)
+    log_to_slack(summary, 'user-accounts') if @num_errors.positive?
+  end
+
+  def accounts_to_scrub
+    accounts = Queries::User::ExpiredDeletedAccounts.call(deleted_before: @scrub_accounts_deleted_since)
+    if accounts.count > @max_accounts_to_scrub
+      raise SafetyConstraintViolation, "Too many accounts to scrub: #{accounts.count} exceeds limit of #{@max_accounts_to_scrub}"
+    end
```

Comment:
`ActiveRecord#count` triggers `SQL COUNT(*)` request each time it is called
```suggestion
    deleted_accounts_count = accounts.count
    if deleted_accounts_count > @max_accounts_to_scrub
      raise SafetyConstraintViolation, "Too many accounts to scrub: #{deleted_accounts_count} exceeds limit of #{@max_accounts_to_scrub}"
    end
```